File: FloatItPhone/src/main/java/net/geekstools/floatshort/PRO/Util/OpenApplicationsLaunchPad.java
/*
 * Copyright © 2019 By Geeks Empire.
 *
 * Created by Elias Fazel on 11/11/19 7:18 PM
 * Last modified 11/11/19 7:16 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO.Util;

import android.app.Activity;
import android.os.Bundle;

import net.geekstools.floatshort.PRO.Util.Functions.FunctionsClass;

public class OpenApplicationsLaunchPad extends Activity {

    FunctionsClass functionsClass;

    @Override
    protected void onCreate(Bundle Saved) {
        super.onCreate(Saved);
        functionsClass = new FunctionsClass(getApplicationContext(), OpenApplicationsLaunchPad.this);
        String packageName = getIntent().getStringExtra("packageName");
        try {
            if (getIntent().hasExtra("className")) {
                String className = getIntent().getStringExtra("className");
                functionsClass.openApplicationFromActivity(packageName, className);
            } else {
                functionsClass.openApplicationFromActivity(packageName);
            }
        } catch (Exception e) {
            e.printStackTrace();
            functionsClass.openApplicationFromActivity(packageName);
        }
        finish();
    }
}

File: FloatingMultitasking/src/main/java/net/geekstools/floatshort/PRO/Utils/LaunchPad/OpenApplicationsLaunchPad.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 8/29/20 3:58 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.floatshort.PRO.Utils.LaunchPad

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClassLegacy

class OpenApplicationsLaunchPad : AppCompatActivity() {

    override fun onCreate(saved: Bundle?) {
        super.onCreate(saved)

        val functionsClass = FunctionsClassLegacy(applicationContext)

        val appPackageName: String? = intent.getStringExtra("PackageName")

        appPackageName?.let {

            try {
                if (intent.hasExtra("ClassName")) {

                    val appClassName = intent.getStringExtra("ClassName")

                    functionsClass.openApplicationFromActivity(this@OpenApplicationsLaunchPad,
                            appPackageName,
                            appClassName)

                } else {

                    functionsClass
                            .openApplicationFromActivity(this@OpenApplicationsLaunchPad,
                                    appPackageName)

                }
            } catch (e: Exception) {
                e.printStackTrace()

                functionsClass
                        .openApplicationFromActivity(this@OpenApplicationsLaunchPad,
                                appPackageName)
            }
        }

        this@OpenApplicationsLaunchPad.finish()
    }

}